base/array.jl

```diff
@@ -251,6 +251,14 @@ for (fname, felt) in ((:zeros,:zero), (:ones,:one))
         $fname(dims::Tuple) = ($fname)(Float64, dims)
         $fname(T::Type, dims...) = $fname(T, dims)
         $fname(dims...) = $fname(dims)
+
+        # #19635
+        function ($fname){T}(::Type{T}, arr::Array{T})
```

Comment:
Hmm, that is a bit tricky. Still, I think I'd favor moving this to `deprecated.jl` and then writing the test like this:
```julia
@test_throws ErrorException error("some error") # TODO: change this to MethodError when 0.6 deprecations are deleted
```